### DISCUSSION TITLE
Github的这个Action配置是什么意思？

### DISCUSSION THREAD

**Discussion by guanpj** [oldwinter/knowledge-garden#61] [提问] [answered]
问下大佬，你的 quartz 库里的这个  Action 配置是什么意思？
      - name: config1 
        run: rm -rf content/.obsidian content/cedict_ts.u8 content/Extras/Templates  && mv content/*.md content/Atlas && find content/ -name "*.md" | xargs -I file  mv -f file content &&  mv content/AboutTheGarden.md content/_index.md 

      - name: config2
        run: "ls content/ && grep -lr --null 'title' content/* | xargs -0 sed -i -E -r 's/title: (.*)/title: \"\\1\"/g'"

      - name: config3 
        run: rm -rf content/*.md-E

mv content/AboutTheGarden.md content/_index.md 这个我知道是设置主页入口。
但是其他的没看懂：
mv content/*.md content/Atlas 为什么要把文章移动到这个文件夹里面；
find content/ -name "*.md" | xargs -I file  mv -f file content 这个没看懂；
config2 和 config3 没看懂用来干什么的。

希望能帮忙解答一下，谢谢！

**Comment by oldwinter** [comment, marked as answer]
我的md源文件，通过linter，自动往frontmatter里面添加title字段。但quartz要求有title字段，并且有双引号括起来，所以这里就是用正则匹配批量加双引号其次应该就是把所有文件移动到根目录，不然要把双中括号的wikilink形式加上路径，显得不够优雅。

在 2023年3月21日星期二，guanpj ***@***.***> 写道：

**Reply by guanpj** [reply]
好的，感谢！
